build/inline_acceleration/firewall/README.md

```diff
@@ -0,0 +1,226 @@
+		-chardev socket,id=chr0, path=/tmp/vhost-user0 \
+		-chardev socket,id=chr1, path=/tmp/vhost-user2 \
+		-netdev vhost-user,id=net0, chardev=chr0, queues=1 \
+		-netdev vhost-user,id=net1, chardev=chr1, queues=1 \
+		-device virtio-net-pci,netdev=net0, mac=00:e8:ca:11:aa:01 \
+		-device virtio-net-pci,netdev=net1, mac=00:e8:ca:11:cc:01 \
+		-vnc :3 &
+	```
+
+	VM2:
+
+	```
+	qemu-system-x86_64 \
+		-enable-kvm -m 8G -smp cores=10, threads=1, sockets=1 -cpu host \
+		-numa node,memdev=mem -mem-prealloc \
+		-object memory-backend-file, id=mem,size=8G, mem-path=/dev/hugepages,share=on \
+		-hda /root/vm2.img \
+		-netdev user,id=nttsip,hostfwd=tcp::10023-:22 -device e1000, netdev=nttsip \
+		-chardev socket,id=chr0, path=/tmp/vhost-user1 \
+		-chardev socket,id=chr1, path=/tmp/vhost-user3 \
+		-netdev vhost-user,id=net0, chardev=chr0, queues=1 \
+		-netdev vhost-user,id=net1, chardev=chr1, queues=1 \
+		-device virtio-net-pci,netdev=net0, mac=00:e8:ca:11:bb:01 \
+		-device virtio-net-pci,netdev=net1, mac=00:e8:ca:11:dd:01 \
+		-vnc :2 &
+	```
+
+4. Configure Firewall pipeline config and config rule on host OVS.
+	* Set pipe:
+	```
+	sudo ovs-p4ctl set-pipe br0 /root/ipdk/build/inline_acceleration/firewall/example/firewall.pb.bin /root/ipdk/build/inline_acceleration/firewall/example/p4Info.txt
+	```
+	(See [Section 4](#section-4:-generating-dependent-files-from-p4c-and-ovs-pipeline-builder) for how to create these input files from P4)
+
+	* Configure Default forwarding rule on host:
+	```
+	sudo ovs-p4ctl add-entry br0 ingress.egress_default "istd.ingress_port=3,action=ingress.carry_port2(2)"
+	```
+
+5. login to VMs using ssh:
+
+	VM1:
+	```
+	ssh -p 10022 root@localhost
+	```
+
+	VM2:
+	```
+	ssh -p 10023 root@localhost
+	```
+
+6. Setup hugepages, PCI addresses in both VM’s
+
+	```
+	echo 1 > /sys/module/vfio/parameters/enable_unsafe_noiommu_mode
+	modprobe vfio-pci
+	echo 3000 > /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages
+	cd /root && ./dpdk/usertools/dpdk-devbind.py -b vfio-pci 00:04.0 00:05.0
+	```
+
+7. copy /root/ipdk/build/inline_acceleration/firewall/example/p4info.txt in VM2 at /root/
+
+8. Run VPP in VM2:
+
+	```
+	export no_proxy=localhost, <address of host>
+	./build-root/install-vpp-native/vpp/bin/vpp -c /root/startup.conf
+	```
+
+9. For testing without offload
+
+	Run ```rte_flow_wrapper disable``` in VPP shell (VM2)
+
+10. Run pktgen-dpdk in VM1:
+
+	```
+	cd Pktgen-dpdk/
+	./Builddir/app/pktgen -l 3-6 -a 00:04.0 -a 00:05.0 -- -T -P -m "5.0,6.1"
+	```
+
+Run the traffic from pktgen in VM1, traffic can be seen coming on port2. For testing traffic with offload stop traffic and re-run vpp application. Run the traffic from VM1, traffic can be seen coming on port2 with higher rate. After this successful run, the traffic rate should have increased because of rules offloaded on p4-ovs.
+
+## Section 4: Generating dependent files from P4C and OVS pipeline builder
+
+Clone and Build - https://github.com/p4lang/p4c.git
```

Comment:
If you up-level the build scripts from `build/networking`, you could likely make that common across the two projects.

Reply:
done by the up-level script as suggested.